uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -327,229 +373,351 @@ protected UpdateBase(final boolean aliasFeatureEnabled) {
+                    shouldRejectUpdate(zone1, providerInZone1, HttpStatus.CONFLICT);
+                }
 
-            @Test
-            void shouldReject_OnlyAliasIdSet_CustomZone() throws Exception {
-                shouldReject_OnlyAliasIdSet(customZone);
-            }
+                @Test
+                void shouldReject_AliasZidSetToSameZone_UaaZone() throws Exception {
+                    shouldReject_AliasZidSetToSameZone(IdentityZone.getUaa());
+                }
 
-            private void shouldReject_OnlyAliasIdSet(final IdentityZone zone) throws Exception {
-                final IdentityProvider<?> idp = buildOidcIdpWithAliasProperties(zone.getId(), null, null);
-                final IdentityProvider<?> createdProvider = createIdp(zone, idp);
-                assertThat(createdProvider.getAliasZid()).isBlank();
-                createdProvider.setAliasId(UUID.randomUUID().toString());
-                shouldRejectUpdate(zone, createdProvider, HttpStatus.UNPROCESSABLE_ENTITY);
-            }
+                @Test
+                void shouldReject_AliasZidSetToSameZone_CustomZone() throws Exception {
+                    shouldReject_AliasZidSetToSameZone(customZone);
+                }
 
-            @ParameterizedTest
-            @MethodSource("shouldReject_ChangingAliasPropertiesOfIdpWithAlias")
-            void shouldReject_ChangingAliasPropertiesOfIdpWithAlias_UaaToCustomZone(
-                    final String newAliasId,
-                    final String newAliasZid
-            ) throws Throwable {
-                shouldReject_ChangingAliasPropertiesOfIdpWithAlias(newAliasId, newAliasZid, IdentityZone.getUaa(), customZone);
+                private void shouldReject_AliasZidSetToSameZone(final IdentityZone zone) throws Exception {
+                    final IdentityProvider<?> idp = createIdp(
+                            zone,
+                            buildOidcIdpWithAliasProperties(zone.getId(), null, null)
+                    );
+                    idp.setAliasZid(zone.getId());
+                    shouldRejectUpdate(zone, idp, HttpStatus.UNPROCESSABLE_ENTITY);
+                }
             }
 
-            @ParameterizedTest
-            @MethodSource("shouldReject_ChangingAliasPropertiesOfIdpWithAlias")
-            void shouldReject_ChangingAliasPropertiesOfIdpWithAlias_CustomToUaaZone(
-                    final String newAliasId,
-                    final String newAliasZid
-            ) throws Throwable {
-                shouldReject_ChangingAliasPropertiesOfIdpWithAlias(newAliasId, newAliasZid, customZone, IdentityZone.getUaa());
-            }
+            @Nested
+            class ExistingAlias {
+                @Test
+                void shouldAccept_OtherPropertiesOfIdpWithAliasAreChanged_UaaToCustomZone() throws Exception {
+                    shouldAccept_OtherPropertiesOfIdpWithAliasAreChanged(IdentityZone.getUaa(), customZone);
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.